src/SOS/Strike/strike.cpp

```diff
@@ -13534,8 +13539,11 @@ class ClrStackImpl
                 }
                 else
                 {
+                    frameNumber++;
+
                     out.WriteColumn(1, InstructionPtr(ip));
-                    out.WriteColumn(2, MethodNameFromIP(ip, bSuppressLines, bFull, bFull));
+                    // To get the correct line number, the ip needs to be adjusted for leaf frames.
```

Comment:
Same request for more detailed commenting : )
a) what exactly do we expect the IP to be pointing to
b) why is the 'correct' line number not the one that maps to the IP we are reporting right next to it
c) why are we only changing the source mapping for leaf frame but not any other frame
d) what is the correlation between internal frames and whether or not we adjust the mapping